packages/liquidator/src/liquidator.js

```diff
@@ -118,18 +121,65 @@ class Liquidator {
           // Override must be one of the default logging levels: ['error','warn','info','http','verbose','debug','silly']
           return Object.values(overrides).every(param => Object.keys(this.logger.levels).includes(param));
         }
+      },
+      whaleDefenseFundWei: {
+        // by default make this disabled
+        value: undefined,
+        isValid: x => {
+          if (x === undefined) return true;
+          return this.toBN(x).gte(this.toBN("0"));
+        }
+      },
+      defenseActivationPercent: {
+        value: undefined,
+        isValid: x => {
+          if (x === undefined) return true;
+          return parseFloat(x) >= 0 && parseFloat(x) <= 100;
+        }
       }
     };
 
     // Validate and set config settings to class state.
-    Object.assign(this, createObjectFromDefaultProps(config, defaultConfig));
+    const configWithDefaults = createObjectFromDefaultProps(config, defaultConfig);
+    Object.assign(this, configWithDefaults);
+
+    // generalize log emitter, use it to attach default data to all logs
+    const log = (severity = "info", data = {}) => {
+      // would rather just throw here and let index.js capture and log, but
+      // its currently not set up to add in the additional context for the error
+      // so it has to be done here.
+      if (logger[severity] == null) {
```

Comment:
ideally this could be just thrown as an error, and bubbled up to index.js. Error stack would give context, but you could additionally have a single error handler in index which pulls out context state from the liquidator and adds it into the log. With that type of setup you could just throw all errors without caring about how to log them.